### ISSUE TITLE
Using MRTK Keyboard for passwords add the password to autocomplete

### ISSUE THREAD

**Issue by Xan2063** [microsoft/MixedRealityToolkit-Unity#8723]
## Describe the bug

We use the MRTK Keyboard to authenticate users with their username and passwort on the Hololens 2. After entering the password once, the password shows up in the autocomplete suggestions in plain text

## To reproduce

1. Use MRTK Keyboard to type text to some input field that is unknown to the autocomplete sytem into an input field (Most passwords will do)
2. Change to another input field
3. Autocomplete shows the password or types of it as a suggestion

## Expected behavior

There has to be a way to prevent certain fields or instances of the Keyboard to add things to autocomplete. Also it should be possible to delete such entries from autocomplete, when they were added previously

## Your setup (please complete the following information)

- Unity Version [ 2019.4]
- MRTK Version [ v2..3.0]

## Target platform (please complete the following information)

- HoloLens
- HoloLens 2

## Additional context

The TouchscreenKeyboard has a secure flag, see signature below. Is this sufficient to suppress the addition to autocomplete?
`/// <summary>   
    ///   <para>Opens the native keyboard provided by OS on the screen.</para>

    /// </summary>   
    /// <param name="text">Text to edit.</param>   
    /// <param name="keyboardType">Type of keyboard (eg, any text, numbers only, etc).</param> 
    /// <param name="autocorrection">Is autocorrection applied?</param> 
    /// <param name="multiline">Can more than one line of text be entered?</param> 
    /// <param name="secure">Is the text masked (for passwords, etc)?</param>
    /// <param name="alert">Is the keyboard opened in alert mode?</param>
    /// <param name="textPlaceholder">Text to be used if no other text is present.</param>
    /// <param name="characterLimit">How many characters the keyboard input field is limited to. 0 = infinite. (Android and iOS only)</param>
    [ExcludeFromDocs]
    public static TouchScreenKeyboard Open(
      string text,
      TouchScreenKeyboardType keyboardType,
      bool autocorrection,
      bool multiline,
      bool secure,
      bool alert,
      string textPlaceholder);
`

**Comment by wiwei** [comment]
Thanks for filing this! This would be a pretty good Help Wanted candidate (i.e. @Xan2063  if you were comfortable adding a public property to this MonoBehaviour that is used to power the Open call - i.e. passing in the secure boolean, that would be a great community PR)

**Comment by Xan2063** [comment]
Could you give me a hint, how I can reset these suggestions? I can implement that but without the reset testing it might be a bit random.

**Comment by vbandi** [comment]
Seems like the issue is either on the Unity or Microsoft side. If you set the ContentType of a TMP_InputField to Password the soft keyboard should open in "secure" mode as per the decompiled TMP_InputField code (ActivateInputFieldInternal method). However, type autocomplete suggestions are still enabled in this case.

**Comment by vbandi** [comment]
Double-checked with the latest Unity and MRTK (not even sure if this is an MRTK issue TBH). 
The simplest repro is to create an empty scene, with a single line of code on Start:

```
public class KeyboardTester : MonoBehaviour
{
    void Start()
    {
        TouchScreenKeyboard.Open("", TouchScreenKeyboardType.Default, false, false, true);
    }
}
```

This sets autocorrection to false, isSecure to true - but when the keyboard that appears and you start typing, autocorrect is on. 
Please help, this is really annoying, not to mention showstopper for security-sensitive apps.

**Comment by Xan2063** [comment]
The latest information I have is that it is a bug in Unity, not honouring the secure flag. It is escalated as a security problem, but that information is already two weeks old.

**Comment by vbandi** [comment]
Thank you @Xan2063 . Do you have a public bug tracker link I could vote and follow?

**Comment by Xan2063** [comment]
I will try to get one. Seems that all il2cpp versions are affected

**Comment by polar-kev** [comment]
@Xan2063, I'll flag this to Unity as well from the MRTK team side.

**Comment by vladlibrestream** [comment]
Awesome can't wait to see this security breach getting fixed =)

**Comment by timkeo** [comment]
Link to the Unity bug: https://issuetracker.unity3d.com/issues/uwp-touchscreenkeyboard-shows-word-suggestions-for-password-and-other-secure-input-fields

The issue is with the [CoreTextEditContext ](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Text.Core.CoreTextEditContext?view=winrt-19041) implementation, were the "secure" flag in Unity's [TouchScreenKeyboard.Open](https://docs.unity3d.com/ScriptReference/TouchScreenKeyboard.Open.html) API isn't properly translated to one of the "Private" [InputScope types](https://docs.microsoft.com/en-us/uwp/api/windows.ui.text.core.coretextinputscope?view=winrt-19041). The bug impacts all UWP builds and not just HoloLens.

I've implemented the fix but it'll take some time to land the backport to 2019 LTS, but since this is a security bug I'll try to expedite the process.

**Comment by MaxWang-MS** [comment]
According to Unity's release notes, this issue has been fixed in 2020.3.7 and 2019.4.26.
